solr/core/src/java/org/apache/solr/core/CoreContainer.java

```diff
@@ -297,6 +297,9 @@ && getZkController().getOverseer() != null
   public static final long INITIAL_CORE_LOAD_COMPLETE = 0x4L;
   private volatile long status = 0L;
 
+  public static final boolean IS_V2_ENABLED =
+      !"true".equals(System.getProperty("disable.v2.api", "false"));
```

Comment:
is this really how we do these system properties in Solr?   Just sort of funny looking starting with `!"true" ;-)

Reply:
Idk - I'm sure there's a handful of ways to do it.  Probably the cleanest way would be to put a static utility method in V2ApiUtils that abstracts the actual mechanism away entirely.

Tbh I just copied this line from an existing place that we lookup this sysprop - it is a draft PR after all 😛. I'll clean up the actual lookup a bit as I get this ready to merge.

Reply:
;-)   I really need to keep a list of all the things that look odd to me that need refactoring ;-)

Reply:
I ultimately pulled these checks into a method: `V2ApiUtils.isEnabled()`